linera-views/src/backends/lru_caching.rs

```diff
@@ -1,45 +1,305 @@
 // Copyright (c) Zefchain Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-//! Add LRU (least recently used) caching to a given store.
+//! Add caching to a given store.
+//! The current policy is based on the LRU (Least Recently Used).
 
-/// The standard cache size used for tests.
-pub const TEST_CACHE_SIZE: usize = 1000;
+use std::{fs::File, io::BufReader};
+
+/// The parametrization of the cache
+#[derive(Clone, Debug, serde::Deserialize)]
+pub struct StorageCachePolicy {
+    /// The maximum size of the cache, in bytes (keys size + value sizes)
+    pub max_cache_size: usize,
+    /// The maximum size of an entry size, in bytes
+    pub max_entry_size: usize,
+    /// The maximum number of entries in the cache.
+    pub max_cache_entries: usize,
+}
+
+/// The maximum number of entries in the cache.
+/// If the number of entries in the cache is too large then the underlying maps
+/// become the limiting factor
+pub const DEFAULT_STORAGE_CACHE_POLICY: StorageCachePolicy = StorageCachePolicy {
+    max_cache_size: 10000000,
+    max_entry_size: 1000000,
+    max_cache_entries: 1000,
+};
+
+/// Read the `StorageCachePolicy` from the existing file and if None, then the default
+/// storage policy is returned
+pub fn read_storage_cache_policy(storage_cache_policy: Option<String>) -> StorageCachePolicy {
+    match storage_cache_policy {
+        None => DEFAULT_STORAGE_CACHE_POLICY,
+        Some(storage_cache_policy) => {
```

Comment:
Why not allow for configuring the caching policy by simply expecting `--max-cache-size <bytes>` , `--max-cache-entries <number>` , etc to the binary? Reading file with these three written down as a JSON seems like an unnecessary complication 🤔 

if you decide to promote them to binary arguments, you can even decide to make `max_cache_size` and `max_cache_entries` mutually exclusive.

Reply:
We generally want to switch to the use of too many command line entries and switch to the use of input in files.